ivy.el

```diff
@@ -3066,6 +3066,18 @@ RE-STR is the regexp, CANDS are the current candidates."
              (not (and ivy--flx-featurep
                        (eq ivy--regex-function 'ivy--regex-fuzzy)
                        (< (length cands) 200)))
+             ;; If there was a preselected candidate, don't try to
+             ;; keep it selected even if the regexp still matches it.
+             ;; See issue #1563.  See also `ivy--preselect-index',
+             ;; which this logic roughly mirrors.
+             (not (let ((preselect (ivy-state-preselect ivy-last))
+                        (current (ivy-state-current ivy-last)))
```

Comment:
Nitpick: again, there are other parts of `ivy--recompute-index` which could make use of these bindings.